dash/include/dash/Lazy.h

```diff
@@ -0,0 +1,77 @@
+#ifndef DASH__LAZY_H__INCLUDED
+#define DASH__LAZY_H__INCLUDED
+
+#include <functional>
+
+
+namespace dash {
+
+/* TODO:
+ * Should be
+ * 
+ *   template <typename T, bool evaluated>
+ *   class Lazy;
+ *
+ * with:
+ *
+ *   constexpr Lazy<T, true>
+ *   Lazy<T, false>::operator T() const {
+ *     return Lazy<T, true>(gen_fun());
+ *   }
+ *
+ * and:
+ *
+ *   constexpr Lazy<T, true>
+ *   Lazy<T, true>::operator T() const {
+ *     return _value;
+ *   }
+ * 
+ *
+ */
+
+template <typename T>
+class Lazy {
+
```

Comment:
For my HDF5 asio streams I already introduces `dash::launch::async` sowie `dash::launch::sync`. We should try to avoid naming similar features differently.